### DISCUSSION TITLE
Support additional Column and Field attributes

### DISCUSSION THREAD

**Discussion by ggabriel96** [tiangolo/pydantic-sqlalchemy#92] [Questions] [answered]
Hi there!

I just wanted to know if there's interest in adding support for additional properties of [Pydantic fields](https://pydantic-docs.helpmanual.io/usage/schema/#field-customisation) from [SQLAlchemy's `Column`](https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.Column). I would want that :sweat_smile: 

There's issue #7 that is related. Additionally, theres `Column.doc` and `Column.comment` that could populate `Field.description`.
One example use case is to improve the automatic documentation when using FastAPI, since the description is used to describe schemas in OpenAPI spec.

Here's an example, taken from the default content in https://editor.swagger.io/:

![image](https://user-images.githubusercontent.com/4575887/117089447-fc1ead80-ad2b-11eb-9692-2aa7172d9a29.png)

That `pet status in the store` is the description of the `status` field.

We could decide what should be supported and how. I _think_ I could come up with a PR if that's the case.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the patience with my reply! :sweat_smile:

I deprecated this library a while ago in favor of SQLModel which would solve all the same use cases in a much better way, and I'm directing all the effort in that direction.

So, I wouldn't add new features (that need to be maintained), including new types.

SQLModel uses Pydantic models to define both data and table (SQL) models, so it's easier to do all these things. :rocket:
